src/Razor/test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/RazorDirectivesTest/BuiltInDirectiveDoesNotErorrIfNotAtStartOfLineBecauseOfWhitespace.stree.txt

```diff
@@ -3,7 +3,7 @@ RazorDocument - [0..26)::26 - [LF  @addTagHelper "*, Foo"]
         MarkupTextLiteral - [0..2)::2 - [LF] - Gen<Markup> - SpanEditHandler;Accepts:Any
             NewLine;[LF];
         CSharpCodeBlock - [2..26)::24
-            CSharpStatementLiteral - [2..4)::2 - [  ] - Gen<Stmt> - SpanEditHandler;Accepts:Any
+            CSharpEphemeralTextLiteral - [2..4)::2 - [  ] - Gen<Stmt> - SpanEditHandler;Accepts:Any
```

Comment:
Someday you'll have to explain to me what this is... My intuition is that this **should** be considered trivia.

Reply:
> Someday you'll have to explain to me what this is

I'll explain right now. Consider,
```
@{
    @@if (true) { }
    ^
}
```
In this case, even though we have two `@`s, we only want to one of them to be rendered in the output. It is a CSharp Statement but it should not be output. Hence, CSharpEphemeralTextLiteral exists.
If you are curious how it worked in the old world, this `@` was classified as a code span with `Null` chunk generator.

Similarly we have that case in markup as well,
```
<div>@@foo</div>
     ^
```
We use the extra transition to escape the `@` character. So, that transition should not be rendered in the output. Hence, `MarkupEphemeralTextLiteral` exists.


> My intuition is that this should be considered trivia.

I agree that **this particular case**(whitespace preceding a directive) is a good case for trivia but that doesn't mean we can get rid of the ephemeral text literals.